src/test/java/com/google/api/codegen/testdata/go_main_library.baseline

```diff
@@ -688,13 +688,13 @@ func (it *BookIterator) PageInfo() *iterator.PageInfo {
 
 // Next returns the next result. Its second return value is iterator.Done if there are no more
 // results. Once Next returns Done, all subsequent calls will return Done.
-func (it *BookIterator) Next() (*librarypb.Book, error) {
-    if err := it.nextFunc(); err != nil {
-        return nil, err
+func (it *BookIterator) Next() (item *librarypb.Book, err error) {
```

Comment:
Your change to the generator is fine by me, but you could also write the code this way:

```
func (it *BookIterator) Next() (*librarypb.Book, error) {
    var item *libraryp.Book
    if err := it.nextFunc(); err != nil {
        return item, err
    }
    item = it.items[0]
    it.items = it.items[1:]
    return item, nil
}
```
I would much prefer that.